crema/parsers/comet.py

```diff
@@ -0,0 +1,209 @@
+        sequences and a second colun labeled 'decoy' that contains decoy
+        sequences. This file can be generated by setting --peptide-list=T
+        in tide-index.
+    copy_data : bool, optional
+        If true, a deep copy of the data is created. This uses more memory, but
+        is safer because it prevents accidental modification of the underlying
+        data. This argument only has an effect when `txt_files` is a
+        :py:class:`pandas.DataFrame`
+
+    Returns
+    -------
+    PsmDataset
+        A :py:class:`~crema.dataset.PsmDataset` object containing the parsed
+        PSMs.
+    """
+    target = "target/decoy"
+    peptide = "modified_peptide"
+    spectrum = ["scan", "exp_neutral_mass"]  # No file col to use
+    pairing = ""
+    protein = "protein"
+    protein_delim = ","
+
+    # Possible score columns output by Comet.
+    scores = {
+        "e-value",
+        "xcorr",
+        "sp_score",
+    }
+    scores_all = scores
+
+    # Keep only Comet scores that exist in all of the files.
+    skip_first_line = False
+    crux_comet = False  # tracks whether standalone comet or Crux Comet
+    if isinstance(txt_files, pd.DataFrame):
+        scores = scores.intersection(set(txt_files.columns))
+    else:
+        txt_files = utils.listify(txt_files)
+        for txt_file in txt_files:
+            with open(txt_file) as txt_ref:
+                # First line of Comet output consists only of version
+                # If statement below in case first line is removed
+                line = txt_ref.readline().rstrip()
+                if line.startswith("CometVersion"):
+                    line = txt_ref.readline().rstrip()
+                    skip_first_line = True
+                cols = line.split("\t")
+                scores = scores.intersection(set(cols))
+
+                # Standalone Comet and Crux Comet use different column headers
+                if protein not in cols:
+                    protein = "protein id"
+                    peptide = "modified sequence"
+                    spectrum = ["scan", "spectrum neutral mass"]
+                    crux_comet = True
+
+    if not scores:
+        raise ValueError(
+            "Could not find any of the Comet score columns in all of the files."
+            f" The columns Crema looks for are {', '.join(list(scores_all))}"
+        )
+
+    scores = list(scores)
+
+    # Read in the files:
+    fields = spectrum + [peptide] + [target] + scores + [pairing] + [protein]
+    if isinstance(txt_files, pd.DataFrame):
+        data = txt_files.copy(deep=copy_data).loc[:, fields]
+    else:
+        data = pd.concat(
+            [
+                utils.parse_psms_txt(f, fields, skip_first_line)
+                for f in txt_files
+            ]
+        )
+
+    if crux_comet:
+        data["target/decoy"] = ~data[protein].str.contains("decoy_")
+    else:
+        data["target/decoy"] = ~data[protein].str.contains("DECOY_")
```

Comment:
**question:** Isn't the decoy prefix a search engine parameter in both cases? If that's true, then we can't count on the protein being prefixed with "decoy_" or "DECOY_" exclusively.

Reply:
Good point. What I have here are the defaults values. Makes sense to add an argument in case it case it is different.